src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagLinkedList.cs

```diff
@@ -149,6 +150,121 @@ public void AddFront(T value)
         public DiagEnumerator<T> GetEnumerator() => new DiagEnumerator<T>(_first);
         IEnumerator<T> IEnumerable<T>.GetEnumerator() => GetEnumerator();
         IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+
+        private static void ActivityLinkToString(ref ActivityLink al, ref ValueStringBuilder vsb)
+        {
+            ActivityContext ac = al.Context;
+
+            vsb.Append("(");
+            vsb.Append(ac.TraceId.ToHexString());
+            vsb.Append(", ");
+            vsb.Append(ac.SpanId.ToHexString());
+            vsb.Append(", ");
+            vsb.Append(ac.TraceFlags.ToString());
+            vsb.Append(", ");
+            vsb.Append(ac.TraceState ?? "null");
+            vsb.Append(", ");
+            vsb.Append(ac.IsRemote ? "true" : "false");
+
+            if (al.Tags is not null)
+            {
+                vsb.Append(", [");
+                string sep = "";
+                foreach (KeyValuePair<string, object?> kvp in al.EnumerateTagObjects())
+                {
+                    vsb.Append(sep);
+                    vsb.Append(kvp.Key);
+                    vsb.Append(": ");
+                    vsb.Append(kvp.Value?.ToString() ?? "null");
```

Comment:
This could be tricky to parse out. Because `kvp.Value` could be a string which includes `:` and\or `,` characters.

Reply:
What do you suggest?

Reply:
@CodeBlanch By the way, we format the TagObjects today the same. right?

Reply:
What do you think if use `NARROW NO-BREAK SPACE (U+202F): A narrow space character that prevents line breaks.` instead of the normal space? would that help better in parsing?

Reply:
> By the way, we format the TagObjects today the same. right?

Ya you're right. Here is the code I have to parse the existing ones: https://github.com/dotnet/diagnostics/blob/110423bf0805511ee52da601f132dbf01410dad7/src/Microsoft.Diagnostics.Monitoring.EventPipe/DistributedTraces/TraceEventExtensions.cs#L171

I guess fine to do the same thing here. Buyer beware 😄

Reply:
> What do you think if use Non-printable Unicode characters, such as Zero-width space (\u200B) instead of the normal space? would that help better in parsing?

Yes, this could help with parsing in some cases.

Reply:
@rajkumar-rangaraj I have now using Zero-width space (\u200B). You may look at the description of the PR above to see how it will look like.